### PR TITLE
NewVolumeManager expects nodeName instead of hostName. 

### LINKED ISSUE #28643: Attach detach controller not working in k8s 1.3 for AWS EBS volume
I upgraded to the latest k8s 1.3 and trying to leverage the latest attach detach controller. But I'm unable to mount AWS EBS volumes. I get the below error:

Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "mypod-2170121488-bh7ir"/"kube-system". list of unattached/unmounted volumes=[my-storage]

As per AWS, the volume status is in-use and attached but k8s is unable to mount it correctly on the pod as specified in the error above.

### PR DESCRIPTION
**Pull request by dhawal55**
Fixes #28643

### CODE REVIEW

**Comment by justinsb** [comment]
@dhawal55 we already have a fix for the duplicate initialization in progress apparently (#29234), and I also think we'll cherry-pick the nodename/hostname confusion, but possibly not the double-init (depending on how problematic it is).  Can you make this just the one-liner fix - will make it much easier to cherry-pick?

This presumes you can sign the CLA though!  If not let me know and I can just extract the same (?) patch out of my PR.

**Comment by justinsb** [comment]
@saad-ali this fixes #28643.  Feel free to assign to me if you want.  Also I did #29048 which introduces a strong type for the nodename to check for other occurrences of this, and I think we're otherwise OK.

And so @dhawal55 can you please also put "Fixes #28643" into the commit message body.  (And the first line should be shorter, but that should happen anyway without the duplicate code removal).

**Comment by dhawal55** [comment]
Any idea what this means - "We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for the commit author(s).". There is only one commit and its by me.

**Comment by dhawal55** [comment]
@justinsb I submitted this pull request to master. Still trying to understand k8s release process. Should i submit the pull request to "Release 1.3" branch?

**Comment by pmorie** [comment]
@dhawal55 would you edit the description of this PR to explain what this PR changes and why it is needed?  Thank you.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -522,7 +507,7 @@ func NewMainKubelet(
 
 	klet.volumeManager, err = volumemanager.NewVolumeManager(
 		enableControllerAttachDetach,
-		hostname,
+		nodeName,
```
Comment (pmorie) [inline pkg/kubelet/kubelet.go]: What else does this affect?

**Comment by saad-ali** [comment]
> And so @dhawal55 can you please also put "Fixes #28643" into the commit message body. (And the first line should be shorter, but that should happen anyway without the duplicate code removal).

Change LGTM. But as @justinsb mentioned please change the commit title/message (not the PR title/message). You can do this by:

1. git rebase -i HEAD~1
2. Replace `pick` with `r` for your commit, save and quit.
3. A new editor will start allowing you to edit the commit title and message for your commit.
4. Modify, save, and quit.

> @justinsb I submitted this pull request to master. Still trying to understand k8s release process. Should i submit the pull request to "Release 1.3" branch instead/also?

No this is fine. We first merge to head. Once that is done we create a new cherry pick to release-1.3.

**Comment by saad-ali** [comment]
Also can you address the CLA bot:

> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for the commit author(s). If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA (login here to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (dhawal55) [inline pkg/kubelet/kubelet.go]: The volume manager expects a nodeName and uses it to query the API server to verify that the attach detach controller has attached the volume to the node. When the hostname does not matches the nodeName, this lookup fails eventually failing volume mounting.

Comment (dhawal55) [inline pkg/kubelet/kubelet.go]: I haven't seen any other side affect as the nodeName is only used in reconciler and to build cache of current state (volumes and pods that mount them)

**Comment by dhawal55** [comment]
@saad-ali Thanks for the detailed steps on how to update commit message. I have updated both commit message and author to match the CLA. However, the CLA check is still failing.

**Comment by saad-ali** [comment]
@dhawal55 Can you confirm you pushed the changes? The old commit still shows in this PR.

Make sure that you have upstream/origin correctly. Run `git remote -v` to verify.

In your clone, set your origin/upstream as follows:
  * `git remote add upstream https://github.com/ORIGINAL_OWNER/ORIGINAL_REPOSITORY.git`
  * `git remote set-url origin https://github.com/YOUR_USERNAME/YOUR_FORK.git`
  * To prevent accidental pushes to the original repo:
    * `git remote set-url --push upstream no_push`
    * `git remote -v` to check everything.

**Comment by dhawal55** [comment]
@saad-ali duh!! I had 2 very similar named branches and i was updating the wrong one. Thanks for bearing with me.

**Comment by saad-ali** [comment]
No problem. Thanks for the fix. LGTM

**Comment by saad-ali** [comment]
@k8s-bot ok to test

**Comment by saad-ali** [comment]
@k8s-bot node e2e test this issue: #29395

**Pull request merged**